src/vgg.jl

```diff
@@ -0,0 +1,248 @@
+  layers = []
+  push!(layers, flatten)
+  push!(layers, Dense(Int(prod(imsize)), fcsize, relu))
+  push!(layers, Dropout(dropout))
+  push!(layers, Dense(fcsize, fcsize, relu))
+  push!(layers, Dropout(dropout))
+  push!(layers, Dense(fcsize, nclasses))
+
+  return layers
+end
+
+"""
+    vgg(imsize; config, inchannels, batchnorm=false, nclasses, fcsize, dropout)
+
+Create a VGG model
+([reference](https://arxiv.org/abs/1409.1556v6)).
+
+# Arguments
+- `imsize`: input image width and height as a tuple
+- `config`: the configuration for the convolution layers
+            (see [`Metalhead.vgg_convolutional_layers`](#))
+- `inchannels`: number of input channels
+- `batchnorm`: set to `true` to use batch normalization after each convolution
+- `nclasses`: number of output classes
+- `fcsize`: intermediate fully connected layer size
+            (see [`Metalhead.vgg_classifier_layers`](#))
+- `dropout`: dropout level between fully connected layers
+"""
+function vgg(imsize; config, inchannels, batchnorm=false, nclasses, fcsize, dropout)
+  conv = vgg_convolutional_layers(config, batchnorm, inchannels)
+  imsize = outputsize(conv, (imsize..., inchannels); padbatch=true)[1:3]
+  class = vgg_classifier_layers(imsize, nclasses, fcsize, dropout)
+  return Chain(conv..., class...)
+end
+
+const vgg_config = Dict(:A => [(64,1), (128,1), (256,2), (512,2), (512,2)],
+                        :B => [(64,2), (128,2), (256,2), (512,2), (512,2)],
+                        :D => [(64,2), (128,2), (256,3), (512,3), (512,3)],
+                        :E => [(64,2), (128,2), (256,4), (512,4), (512,4)])
+
+"""
+    VGG{BN}(imsize=(224, 224); config, inchannels, nclasses, fcsize, dropout)
+
+Create a `VGG` model
+([reference](https://arxiv.org/abs/1409.1556v6)).
+See also [`vgg`](#).
+
+Set `BN` to `true` to enable batch normalization after
+convolution layers. Defaults to `false`.
+
+# Arguments
+- `imsize`: input image width and height as a tuple
+- `config`: the configuration for the convolution layers
+            (see [`Metalhead.vgg_convolutional_layers`](#))
+- `inchannels`: number of input channels
+- `nclasses`: number of output classes
+- `fcsize`: intermediate fully connected layer size
+            (see [`Metalhead.vgg_classifier_layers`](#))
+- `dropout`: dropout level between fully connected layers
+"""
+struct VGG{BN, T}
+  layers::T
+end
+
+function VGG{BN}(imsize=(224, 224); config, inchannels, nclasses, fcsize, dropout) where BN
+  layers = vgg(imsize; config=config,
+                        inchannels=inchannels,
+                        nclasses=nclasses,
+                        fcsize=fcsize,
+                        dropout=dropout,
+                        batchnorm=BN)
+  
+  VGG{BN, typeof(layers)}(layers)
+end
+
+Functors.functor(::Type{VGG{BN, T}}, m) where {BN, T} =
+  (layers = m.layers,), nt -> VGG{BN, typeof(nt.layers)}(nt.layers)
+
+VGG(imsize=(224, 224); config, inchannels, nclasses, fcsize, dropout) =
```

Comment:
```suggestion
VGG(imsize = (224, 224); config, inchannels, nclasses, fcsize, dropout) =
```

Where is the image size necessary knowledge? It'd be better if users didn't have to specify it at the start. Is it the fully connected layers?

Reply:
Yes, the first FC layer input will be sized differently depending on your data. There's really no way around that fact for VGG. Assuming the user is using VGG for the default dataset of ImageNet, they won't have to specify the image size.

Reply:
Hmm, maybe it's better to separate the FC layers from the rest of the model.

Reply:
The ability to specify image size here is a feature, not really something I want to remove. Why make a user calculate the FC input size when they don't need to?